src/main/java/GameField.java

```diff
@@ -16,22 +16,32 @@ public GameField(Point size){
     }
 
     private boolean isFree(Rectangle rectangle){
-        if (outOfField(rectangle)) return false;
-        if (intersectsObjects(rectangle)) return false;
+        if (outOfField(rectangle)){
+            return false;
+        }
+        if (intersectsObjects(rectangle)){
+            return false;
+        }
         return true;
     }
 
     private boolean intersectsObjectsExcept(Rectangle rectangle, GameObject exception){
         for(GameObject gameObject: gameObjects){
-            if (exception == gameObject) continue;
-            if (gameObject.intersects(rectangle)) return true;
+            if (exception == gameObject){
```

Comment:
comparing two objects by reference. equals()?

Reply:
no, check if it is the same object. Because object always intersects with itself, want to avoid this